pandas/core/arrays/datetimelike.py

```diff
@@ -517,8 +517,12 @@ def __getitem__(self, key):
                 return self._box_func(val)
             return type(self)(val, dtype=self.dtype)
 
+        if is_list_like(key) and not isinstance(key, tuple):
+            key = check_array_indexer(self, key)
+
         if com.is_bool_indexer(key):
-            key = check_bool_array_indexer(self, key)
+            # can still have object dtype
```

Comment:
are there *any* uses of is_bool_indexer left? can you just get rid of them.

Reply:
Yes, we still use it in several places. As long as we don't have deprecated+removed boolean indexing with object dtype (again, see the non-inline discussion in this PR, https://github.com/pandas-dev/pandas/pull/31150#issuecomment-577679134), we will need this

Reply:
can u create an issue to rename / refactor / remove is_book_indexer then
as it’s purpose is now different before - it is no longer the one true way

and now we have 2 ways of checking booking indexers

either check_array_indexer should completely subsume it or it should be renamed

Reply:
> and now we have 2 ways of checking booking indexers

Yes, because there are places we allow object dtype (for backwards compatibility), and there are places where we are more strict.

Why would it need to be renamed? Or what name do you suggest?

Reply:
i would just incorporate it in check_array_indexer to be honest 
it’s more complicated the way you have have it now and just plain confusing 

i have to know that is_bool_indexer is something that doesn’t check indexing except in object arrays